canvas_modules/common-canvas/__tests__/common-properties/actions/button-action-test.js

```diff
@@ -170,51 +191,53 @@ describe("action-button renders correctly", () => {
 				"parameter_ref": "number"
 			}
 		};
-		const wrapper = mount(
+		const wrapper = render(
 			<Provider store={controller.getStore()}>
 				<ActionButton
 					action={actionWithoutButtonObject}
 					controller={controller}
 				/>
 			</Provider>
 		);
-		const button = wrapper.find("button");
+		const { container } = wrapper;
+		const contButton = container.getElementsByClassName("cds--btn")[0];
+		const button = wrapper.getAllByRole("button");
 		expect(button).to.have.length(1);
 		// verify default button kind is tertiary
-		expect(button.prop("className").includes("cds--btn--tertiary")).to.equal(true);
+		expect(contButton.className.includes("cds--btn--tertiary")).to.equal(true);
 		// verify default button size is small
-		expect(button.prop("className").includes("cds--btn--sm")).to.equal(true);
+		expect(contButton.className.includes("cds--btn--sm")).to.equal(true);
 	});
 });
 
 describe("actions using paramDef", () => {
 	let wrapper;
 	let renderedObject;
 	beforeEach(() => {
-		renderedObject = propertyUtils.flyoutEditorForm(ACTION_PARAMDEF);
+		renderedObject = propertyUtils.flyoutEditorFormRender(ACTION_PARAMDEF);
 		wrapper = renderedObject.wrapper;
 	});
 
 	it("should fire action when button clicked", (done) => {
-		renderedObject = propertyUtils.flyoutEditorForm(ACTION_PARAMDEF, null, { actionHandler: callback }, { appData: appData });
+		renderedObject = propertyUtils.flyoutEditorFormRender(ACTION_PARAMDEF, null, { actionHandler: callback }, { appData: appData });
 		wrapper = renderedObject.wrapper;
 		function callback(id, inAppData, data) {
 			expect(id).to.equal("increment");
 			expect(inAppData).to.eql(appData);
 			expect(data.parameter_ref).to.equal("number");
 			done();
 		}
-		const button = wrapper.find("div[data-id='increment'] button");
-		button.simulate("click");
+		const button = wrapper.getAllByRole("button")[38];
+		fireEvent.click(button);
 	});
 
```

Comment:
Is it possible to still use data-id / data-test-id wherever we used to check using data-id?

Reply:
I'll look into it

Reply:
there's wrapper.getAllByTestId but it requires the datatest-id = "increment" and not data-id
There's also container.getElementById, but for some reason it's saying that this isn't a valid function. 
Do you have any other ideas?

Reply:
you can try something like below.
```
const { container } = wrapper;
const div = container.querySelector("div[data-id='increment']");
const button = within(div).getByRole('button');
fireEvent.click(button);
```

Reply:
Thank you for that! That works and is much easier to use!